### PR TITLE
feat: implement python-poetry packages

### LINKED ISSUE #1229: python release type not able to find correct python packages when using poetry
**Is your feature request related to a problem? Please describe.**

When using poetry and package name contains dash, python release type not able to find correct version files 

 **Describe the solution you'd like**

Have a better guess by replacing dash with an underscore.

For example, I have a pyproject.toml

```toml
[tool]

[tool.poetry]
name = "my-service-client"
version = "0.1.0"
description = ""
```

Release please warn me
```bash
file my-service-client/__init__.py did not exist
```

while is file path is actually `my_service_client/__init__.py`

 **Describe alternatives you've considered**

Package name and extra files should be able to bump the file correctly, however guessing the package name better is always something nice to have.

### PR DESCRIPTION
**Pull request by goatwu1993**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/release-please/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Implement [packages](https://github.com/ChenJinGit/poetry/blob/50360f2dc96881398b428a08bd9288c4cc447d4e/docs/docs/pyproject.md#packages) of poetry pyproject.toml

Fixes #1229 🦕

### CODE REVIEW

File: src/strategies/python.ts [hunk 1]
```diff
@@ -100,22 +100,38 @@ export class Python extends BaseStrategy {
...
-    if (!projectName) {
+    const packagePaths = new Set<string>();
+    if (
+      parsedPyProject?.tool?.poetry &&
+      parsedPyProject?.tool?.poetry.packages
+    ) {
+      parsedPyProject?.tool?.poetry.packages.forEach(pkg => {
+        // Use 'from' if it appears, else use 'include'.
+        // Skip glob 'extra/**/*.py by split
+        packagePaths.add((pkg.from || pkg.include).split('/')[0]);
```
Comment (chingor13) [inline src/strategies/python.ts:111]: Could the glob be something like `extra/path/**/*.py`? For example, perhaps there are 2 packages nested under `extra`.

File: src/strategies/python.ts [hunk 2]
```diff
@@ -100,22 +100,38 @@ export class Python extends BaseStrategy {
       );
     }
 
-    if (!projectName) {
+    const packagePaths = new Set<string>();
+    if (
+      parsedPyProject?.tool?.poetry &&
+      parsedPyProject?.tool?.poetry.packages
+    ) {
+      parsedPyProject?.tool?.poetry.packages.forEach(pkg => {
```
Comment (chingor13) [inline src/strategies/python.ts:108]: nit: the `?.` let's us chain these together. Then in the block, typescript knows that the fields exist so we can drop the `?.`.

```suggestion
      parsedPyProject?.tool?.poetry?.packages
    ) {
      parsedPyProject.tool.poetry.packages.forEach(pkg => {
```

**Review by chingor13 — CHANGES_REQUESTED** [review]
This is looking pretty good.

Question: these are just extra modules (that have an `__init__.py` file) that are all expected to be at the same version together? Would this prevent someone from using a manifest config that has multiple components at different versions?

**Comment by goatwu1993** [comment]
Hmm. I took a look and seems there is more to consider.

I will make some refactoring, maybe split some poetry toml logic and create a new PR

**Pull request closed without merging**